sale_commission/models/account_invoice.py

```diff
@@ -117,6 +117,15 @@ class AccountInvoiceLineAgent(models.Model):
     settled = fields.Boolean(
         compute="_compute_settled",
         store=True, copy=False)
+    company_id = fields.Many2one(
```

Comment:
Is it needed to add this burden?

Reply:
now checking

Reply:
It is needed otherwise:
```
commission/sale_commission/models/settlement.py", line 198, in _check_company
    if record.agent_line.company_id != record.company_id:
AttributeError: 'account.invoice.line.agent' object has no attribute 'company_id'
```
I added it in its commit message.
Being here `company_id` a computed field I don't think any migration script is needed, but I've added a record rule

Reply:
The problem I see is performance, so balance the need (or other way of getting the same) with this in mind.

Reply:
@pedrobaeza would a related field be more efficient in your opinion?
Because looking at the computing method, that should be enough.

On a side note, the computing method is exactly the same as it is in the following versions, there might be performance issues over there too

Reply:
Well, I was just guessing... if there's a record rule directly associated to this model, then the related stored field is the best option between performance and feature.

Reply:
@pedrobaeza changed to related field